File: Data Structures & Algorithms/level-order-traversal-of-binary-tree/submission-1.js
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     constructor(val = 0, left = null, right = null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    /**
     * @param {TreeNode} root
     * @return {number[][]}
     */
    levelOrder(root) {
        let res = []
        this.bfs(root, 0, [],res)

        return res
    }

    bfs(node, level, subset, res) {
        if (!node) {return} 

        if (res.length <= level) {res.push([])}
        subset.push(node.val)

        res[level].push(subset)

        this.bfs(node.left,level+1, [], res)
        this.bfs(node.right, level+1, [], res)

    }

}

File: Data Structures & Algorithms/binary-tree-from-preorder-and-inorder-traversal/submission-1.go
import ("slices")
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */

func buildTree(preorder []int, inorder []int) *TreeNode {
	if len(preorder) == 0 || len(inorder) == 0 { return nil }

	res := &TreeNode {
		Val: preorder[0],
	}

	idx := slices.Index(inorder, preorder[0])
	if idx == -1 {
		return nil
	}
	res.Left = buildTree(preorder[1:1+idx], inorder[:idx])
	res.Right = buildTree(preorder[1+idx:], inorder[idx+1:])

	return res
}